### PR TITLE
mgr/dashboard_v2: add 404 page

### PR DESCRIPTION
**Pull request by tspmelo**
Add a 404 page that will be triggered every time the user tries to access a route that does not exists or the servers responds with 404.

image

`Signed-off-by: Tiago Melo <tmelo@suse.com>`

### CODE REVIEW

**Comment by tspmelo** [comment]
Maybe I could change the text to say "cephalopod" or have like 10 local images that will be displayed?

File: src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'cd-not-found',
+  templateUrl: './not-found.component.html',
+  styleUrls: ['./not-found.component.scss']
+})
+export class NotFoundComponent implements OnInit {
```
Comment (votdev) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.ts]: If ngOnInit() is not used, then we do not have to implement that interface OnInit.

**Comment by votdev** [comment]
Using an image is a nice idea, but this requires that we need to update it every Ceph major release when we display the octopus of the Ceph release name.

**Review by votdev — DISMISSED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.ts (hunk 1 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.ts]: done.

**Review by votdev — DISMISSED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<div class="row">
+  <div class="col-md-12 text-center">
+    <h1>Sorry, we could not find what you were looking for</h1>
+    <h2>Instead, enjoy some cephalopods!!!</h2>
+
+    <img class="img-responsive center-block"
```
Comment (Devp00l) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.html]: Could you add the class "img-rounded" than the sharp edges are removed and it looks a lot nicer just by doing this.

**Comment by sebastian-philipp** [comment]
I have my doubts that adding an image here is such a great idea. But +1 for adding a 404 page.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.html (hunk 2 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.html]: done

**Review by Devp00l — DISMISSED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<div class="row">
+  <div class="col-md-12 text-center">
+    <h1>Sorry, we could not find what you were looking for</h1>
+    <h2>Instead, enjoy some cephalopods!!!</h2>
```
Comment (LenzGr) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/core/not-found/not-found.component.html]: Per our discussion, let's stick to displaying the static image here and replace it with a new image once a new Ceph release has been made. I suggest to simply remove the `<h2>` then.

**Review by LenzGr — DISMISSED** [review]

**Review by Devp00l — APPROVED** [review]

**Review by ricardoasmarques — APPROVED** [review]

**Pull request merged**